lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -0,0 +1,682 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the vault-kv relation.
+
+This library contains the Requires and Provides classes for handling the vault-kv
+interface.
+
+## Getting Started
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.vault_k8s.v0.vault_kv
+```
+
+### Requirer charm
+The requirer charm is the charm requiring a secret value store. In this example, the requirer charm
+is requiring a secret value store.
+
+```python
+import secrets
+
+from charms.vault_k8s.v0 import vault_kv
+from ops.charm import CharmBase, InstallEvent
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus
+
+NONCE_SECRET_LABEL = "nonce"
+
+
+class ExampleRequirerCharm(CharmBase):
```

Comment:
I'm missing a provider charm example

Reply:
We are planning on refactoring this library, I prefer to delay adding the example if that's not an issue.